### ISSUE TITLE
Combined save&discard dialogue for all elements of a task

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal-web#482]
What I've noticed when using e.g. Input Filters & Arguments was that each of the tabs had to be saved separately. e.g. when I edit the parameters and want to switch to Input filters, the Discard or Save dialogue shows up.

<img width="1624" alt="Screenshot 2024-05-07 at 13 33 18" src="https://github.com/fractal-analytics-platform/fractal-web/assets/18033446/6c807cf5-3a9b-4827-a7c5-dad2bbab8075">

When we're putting in more of this unified save dialogue (see https://github.com/fractal-analytics-platform/fractal-web/issues/197), I start to wonder: Can we have the save/discard logic be applied to the whole task instead of each of its tabs?

No some of the logic would have to change:
I don't think it makes sense to have an endpoint that always overwrites all the parts (e.g. we always update parameters, meta & input filters, even if only arguments changed).
But maybe we can handle it front-end side in that clicking save or discard triggers save or discard for all the tabs if they have unsaved changes?

We'd have to test this a bit with the behavior of "what if I get a validation error during saving, like missing a required argument"?

**Comment by zonia3000** [comment]
Currently the content of each tab is created when the tab is selected and destroyed when the tab changes. To implement a save function that works for the whole task we need to always create all the components and change only their visibility when the tab change. This means having more components in the page and it could increase the page loading time.

Moreover, what about a user having validation errors on multiple tabs? We should find a way to indicate that fixes have to be made on multiple tabs.

**Comment by jluethi** [comment]
> This means having more components in the page and it could increase the page loading time.

Ok, that's another area to consider carefully then. The scaling here may indeed be problematic in workflows with say 20 tasks or so.

> Moreover, what about a user having validation errors on multiple tabs?

Yes, another relevant concern.

**Comment by zonia3000** [comment]
> Ok, that's another area to consider carefully then. The scaling here may indeed be problematic in workflows with say 20 tasks or so.

It doesn't depend on the number of tasks. The components that would need to be loaded together are: Arguments form, Meta properties form, Task info, Input filters and Versions. These are currently created only when needed, so they don't exist when their tab is not selected. Maybe it's not a big overhead, but for sure the workflow page would contain many more components and logic (and I think it is already the page containing more feature of fractal-web).

**Comment by jluethi** [comment]
> I think it is already the page containing more features of fractal-web

Agreed. This issue also isn't an urgent issue, more of the observation that our user flow here is suboptimal. If there are future refactors the simplify the workflow page, let's keep the design goal in mind:

We want to minimize the save or discard dialogues that a user would go through in a normal flow. And such normal flows can contain changing arguments, meta & filters for each task.
